lib/tbot/grpcconn/grpcconn.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package grpcconn
+
+import (
+	"context"
+	"sync"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// errUninitialized is returned when neither a connection or error has been set
+// on the ClientConn. It's the "default" state of an uninitialized connection.
+var errUninitialized = status.Error(codes.Unavailable, "gRPC client connection is uninitialized")
+
+// ClientConn wraps a *grpc.ClientConn to make it possible to create and pass
+// around gRPC client stubs before the connection has been properly established.
+//
+// This is useful in cases where we weren't able to connect and authenticate
+// with the auth server on-startup (e.g. due to a network partition) but might
+// be able to in the future, and it's more useful for tbot to run in a "degraded"
+// state than to exit completely.
+//
+// The zero-value of a ClientConn returns an error with a gRPC Unavailable
+// status code on each RPC.
+type ClientConn struct {
+	mu       sync.RWMutex
```

Comment:
Almost surely not needed.

```suggestion
	mu       sync.Mutex
```